### ISSUE TITLE
KeyError: 'on_type' on BluePrint.apply for FutureGrant

### ISSUE THREAD

**Issue by toadies** [Titan-Systems/titan#63]
When I create a future grant on a resource and keep getting a `KeyError for 'on_type'`.

Reviewing blueprint.py function `generate_manifest`, I noticed special treatment for `FUTURE_GRANT`.  Further inspection, the script is not including the additional required attributes to build a grant sql.

I was able to fix this by commenting out the existing treatment and using this for future grants.
```
if urn in manifest:
     if data != manifest[urn]:
         # raise Exception(f"Duplicate resource {urn} with conflicting data")
         continue
manifest[urn] = data
````

Below error and code
```
grant = FutureGrant(priv="SELECT", on_future_views_in=Schema(name="some_db.some_schema"), to="SOME_DB_SOME_SCHEMA_READ_ONLY_AR")
bp = Blueprint(name="roles")
    bp.add(grant)
    plan = bp.plan(session)
    pprint(plan)
```
Output
```
>>> pprint(plan)
[ResourceChange(action=<Action.ADD: 'add'>,
                urn=URN(urn::#########:future_grant/SOME_DB_SOME_SCHEMA_READ_ONLY_AR?priv=SELECT&on=schema/SOME_DB.SOME_SCHEMA.<VIEW>),
                before={},
                after={'view': ['SELECT']},
                delta={'view': ['SELECT']})]

>>> bp.apply(session, plan)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "venv/lib/python3.10/site-packages/titan/blueprint.py", line 663, in apply
    action_queue = self._compile_plan_to_sql(session_ctx, plan)
  File "venv/lib/python3.10/site-packages/titan/blueprint.py", line 740, in _compile_plan_to_sql
    _queue_change(change)
  File "venv/lib/python3.10/site-packages/titan/blueprint.py", line 728, in _queue_change
    action = lifecycle.create_resource(change.urn, change.after, props)
  File "venv/lib/python3.10/site-packages/titan/lifecycle.py", line 22, in create_resource
    return getattr(__this__, f"create_{urn.resource_label}", create__default)(urn, data, props, if_not_exists)
  File "venv/lib/python3.10/site-packages/titan/lifecycle.py", line 74, in create_future_grant
    on_type = data["on_type"]
KeyError: 'on_type'
```

When reviewing another grant resource GrantOnAll, the following output is provided.
```
>>> grant = GrantOnAll(priv="read", on_type="stage", in_type="schema", in_name="SOME_DB.SOME_SCHEMA", to="SOME_DB_SOME_SCHEMA_READ_ONLY_AR")
>>> bp.add(grant)
>>> plan = bp.plan(session)
>>> pprint(plan)
[ResourceChange(action=<Action.ADD: 'add'>,
                urn=URN(urn::#########:grant_on_all/SOME_DB_SOME_SCHEMA_READ_ONLY_AR?on_type=STAGE&in_type=SCHEMA&in_name=SOME_DB.SOME_SCHEMA),
                before={},
                after={'grant_option': False,
                       'in_name': 'SOME_DB.SOME_SCHEMA',
                       'in_type': 'SCHEMA',
                       'on_type': 'STAGE',
                       'owner': 'SECURITYADMIN',
                       'priv': 'READ',
                       'to': 'SOME_DB_SOME_SCHEMA_READ_ONLY_AR'},
                delta={'grant_option': False,
                       'in_name': 'SOME_DB.SOME_SCHEMA',
                       'in_type': 'SCHEMA',
                       'on_type': 'STAGE',
                       'owner': 'SECURITYADMIN',
                       'priv': 'READ',
                       'to': 'SOME_DB_SOME_SCHEMA_READ_ONLY_AR'})]
```

Using version 0.6.5 w/ python 3.10

**Comment by teej** [comment]
I am able to repro, I'll get it fixed today.

**Comment by teej** [comment]
Should be fixed in [v0.6.8](https://github.com/Titan-Systems/titan/releases/tag/v0.6.8)
